tests/utils.py

```diff
@@ -47,7 +47,13 @@
 from packaging import version
 
 
-async def retry_limited(fun, timeout, *args, **kwargs):
+async def retry_limited(
+    fun: Union[abc.Callable[..., bool], abc.Callable[..., abc.Awaitable[bool]]],
+    timeout: float,
+    interval: float = 0.1,
+    *args: object,
+    **kwargs: object,
+) -> None:
     async def fun_wrapper():
```

Comment:
Maybe we can even replace this whole implementation with a call to the one in src?
```python
    try:
        await util.retry_limited(fun, timeout, interval, *args, **kwargs)
    except asyncio.TimeoutError:
        raise AssertionError("Bounded wait failed")
```

Reply:
Fixed.